e2e/container/kill_test.go

```diff
@@ -0,0 +1,78 @@
+package container
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/cli/e2e/internal/fixtures"
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/gotestyourself/gotestyourself/poll"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKillContainer(t *testing.T) {
+	containerID := runBackgroundTop(t)
+
+	// Kill with a signal should send the signal to the process
+	// if the signal is not a deadly one, the container should still live on
+	result := icmd.RunCmd(
+		icmd.Command("docker", "kill", "-s", "SIGWINCH", containerID),
+	)
+	result.Assert(t, icmd.Success)
+
+	result = icmd.RunCmd(
+		shell(t, "docker inspect -f '{{ .State.Status }}' %s", containerID),
```

Comment:
Maybe poll the check of the status?

Reply:
I shouldn't need to poll here (because the status is supposed to not changed). That said, I wonder if I should even test this signal (i.e. I should just test that the cli sends the correct signal to the daemon) — the handle of signals, etc.. should be on moby integration tests 🤔

Reply:
Yes, I think we can omit this case. "sends the correct signal" can be tested in unit tests, and the rest outside of this repo.